topostats/tracing/dnatracing.py

```diff
@@ -577,87 +603,113 @@ def saveTraceFigures(
-        #     if self.mol_is_circular[dna_num]:
-        #         starting_point = 0
-        #     else:
-        #         starting_point = self.neighbours
-        #     length = len(self.curvature[dna_num])
-        #     plt.plot(self.splined_traces[dna_num][starting_point, 0],
-        #              self.splined_traces[dna_num][starting_point, 1],
-        #              color='#D55E00', markersize=3.0, marker=5)
-        #     plt.plot(self.splined_traces[dna_num][starting_point + int(length / 6), 0],
-        #              self.splined_traces[dna_num][starting_point + int(length / 6), 1],
-        #              color='#E69F00', markersize=3.0, marker=5)
-        #     plt.plot(self.splined_traces[dna_num][starting_point + int(length / 6 * 2), 0],
-        #              self.splined_traces[dna_num][starting_point + int(length / 6 * 2), 1],
-        #              color='#F0E442', markersize=3.0, marker=5)
-        #     plt.plot(self.splined_traces[dna_num][starting_point + int(length / 6 * 3), 0],
-        #              self.splined_traces[dna_num][starting_point + int(length / 6 * 3), 1],
-        #              color='#009E74', markersize=3.0, marker=5)
-        #     plt.plot(self.splined_traces[dna_num][starting_point + int(length / 6 * 4), 0],
-        #              self.splined_traces[dna_num][starting_point + int(length / 6 * 4), 1],
-        #              color='#0071B2', markersize=3.0, marker=5)
-        #     plt.plot(self.splined_traces[dna_num][starting_point + int(length / 6 * 5), 0],
-        #              self.splined_traces[dna_num][starting_point + int(length / 6 * 5), 1],
-        #              color='#CC79A7', markersize=3.0, marker=5)
-        # plt.savefig('%s_%s_splinedtrace_with_markers.png' % (save_file, channel_name))
-        # plt.close()
-
         plt.pcolormesh(self.full_image_data, vmax=vmaxval, vmin=vminval)
         plt.colorbar()
         for dna_num in sorted(self.splined_traces.keys()):
-            plt.plot(self.splined_traces[dna_num][:, 0], self.splined_traces[dna_num][:, 1], color="c", linewidth=1.0)
-        # plt.savefig("%s_%s_splinedtrace.png" % (save_file, channel_name))
+            plt.plot(self.splined_traces[dna_num][:, 0], self.splined_traces[dna_num][:, 1], color="c", linewidth=2.0)
+        # plt.savefig("%s_%s_splinedtrace.png" % (save_file, channel_name), dpi=1000)
         plt.savefig(output_dir / filename / f"{channel_name}_splinedtrace.png")
         LOGGER.info(f"Splined Trace image saved to : {str(output_dir / filename / f'{channel_name}_splinedtrace.png')}")
         plt.close()
 
-        """
-        plt.pcolormesh(self.full_image_data)
+        plt.pcolormesh(self.full_image_data, vmax=vmaxval, vmin=vminval)
+        plt.colorbar()
+        for dna_num in sorted(self.splined_traces.keys()):
+            plt.plot(self.splined_traces[dna_num][:, 0], self.splined_traces[dna_num][:, 1], color='c', linewidth=1.0)
+            if self.mol_is_circular[dna_num]:
```

Comment:
Hi, thanks for spotting this. The useless if statement was there because previously the starting points were different for linear and circular molecules. I have now removed the variable 'starting_point' as it isn't needed anymore